src/pages/index.js

```diff
@@ -0,0 +1,285 @@
+    link: cardModalLinkInput.value,
+  };
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .postInitialCards(inputValues)
+    .then((data) => {
+      previewModalPicDescription.textContent = data.name;
+      previewModalImage.src = data.link;
+      previewModalImage.alt = data.name;
+      const cardElement = getCardElement(data);
+      cardFormElement.reset();
+      cardsList.prepend(cardElement);
+      disableButton(cardSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true);
+    });
+}
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editAvatarInfo(avatarModalLinkInput.value)
+
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      profileAvatar.alt = data.avatar;
+      //closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true);
+    });
+}
+// Delete button
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, !true, "Delete", "Deleting...");
+    });
+}
+function closeModalEscape(event) {
+  if (event.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+function handleOverlay() {
+  allModals.forEach((modal) => {
+    modal.addEventListener("click", (evt) => {
+      if (evt.target === modal) {
+        closeModal(modal);
+      }
+    });
+  });
+}
+cardModalCloseBtn.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+editModalCloseBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+editModalBtn.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(
+    editFormElement,
+    [editModalNameInput, editModalDescriptionInput],
+    settings
+  );
+  openModal(editModal);
+});
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+
+cardModalBtn.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarModalCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+deleteModalCloseBtn.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+deleteForm.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+avatarSaveBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+editSaveBtn.addEventListener("click", () => {
```

Comment:
Please, delete click handlers from the `save buttons`. Any submit handler is supposed to launch a `submit` action for submitting its form. 

You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
Thank you. I removed all click handlers with the save button parameters per your suggestion.